File: Nejman.Newtone.Core/Media/MediaSource.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Nejman.Newtone.Core.Media
{
    public class MediaSource
    {
        public string Artist { get; private set; }
        public string Title { get; private set; }
        public TimeSpan Duration { get; private set; }
        public byte[] Image { get; private set; }
        public string Path { get; private set; }
        public string ID { get; private set; }
        public string ImageUrl { get; private set; }
        public bool IsLocal
        {
            get
            {
                return !Path.StartsWith("https://") && Path.Length > 11;
            }
        }

        public MediaSource(string path = null, string artist = null, string title = null, TimeSpan duration = default, byte[] image = null, string id = null, string imageUrl = null)
        {
            Artist = artist;
            Title = title;
            Duration = duration;
            Image = image;
            Path = path;
            ID = id;
            ImageUrl = imageUrl;
        }
    }
}

File: app/src/main/java/com/nejman/nsec/music_player/media/MediaSource.java
package com.nejman.nsec.music_player.media;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.support.v4.media.MediaMetadataCompat;

import androidx.annotation.NonNull;

import com.nejman.nsec.music_player.MainActivity;
import com.nejman.nsec.music_player.R;

public class MediaSource {
    public String artist;
    public String title;
    public final long duration;
    public Bitmap image;
    public final byte[] imageData;
    public final String path;
    public final String id;
    public final String imageUrl;
    public final String playlistId;
    public final boolean isLocal;

    public MediaSource(String path, String artist, String title, long duration, byte[] image, String id, String imageUrl, String playlistId) {
        this.artist = artist;
        this.title = title;
        this.duration = duration;
        this.imageData = image;

        if (image != null) {
            this.image = BitmapFactory.decodeByteArray(image, 0, image.length);
        }

        this.path = path;
        this.id = id;
        this.imageUrl = imageUrl;
        this.playlistId = playlistId;
        this.isLocal = !path.startsWith("https://") && path.length() > 11;
    }

    public MediaMetadataCompat toMetaData() {
        return MusicPlaybackService.metadataBuilder
                .putString(MediaMetadataCompat.METADATA_KEY_TITLE, this.title)
                .putString(MediaMetadataCompat.METADATA_KEY_ARTIST, this.artist)
                .putLong(MediaMetadataCompat.METADATA_KEY_DURATION, this.duration)
                .putBitmap(MediaMetadataCompat.METADATA_KEY_ART, this.image == null ? BitmapFactory.decodeResource(MainActivity.instance.getResources(), R.drawable.empty_track) : this.image)
                .putString(MediaMetadataCompat.METADATA_KEY_MEDIA_ID, this.id).build();
    }

    @NonNull
    @Override
    public MediaSource clone() {
        return new MediaSource(path, artist, title, duration, imageData, id, imageUrl, playlistId);
    }
}
